[DO NOT MERGE][WIP] Add new Save page box

danacotoran opened the pull request:
**T B C**

AlanGabbianelli commented on config/locales/en.yml:
```diff
@@ -103,6 +103,13 @@ en:
+      add_page_button: Add to your saved pages
+      remove_page_button: Remove from your saved pages
+      see_saved_pages_signed_in: <a href="/asdf-am-signedin" class="govuk-link">See your saved pages</a>
+      see_saved_pages_signed_out: <a href="/asdf-am-signedout" class="govuk-link">Sign in</a> to see your saved pages.
```
What is `/asdf-am-signedout`?

AlanGabbianelli commented on config/locales/en.yml:
```diff
@@ -103,6 +103,13 @@ en:
+      page_was_saved_heading: You’ve saved this page
+      add_page_button: Add to your saved pages
+      remove_page_button: Remove from your saved pages
+      see_saved_pages_signed_in: <a href="/asdf-am-signedin" class="govuk-link">See your saved pages</a>
+      see_saved_pages_signed_out: <a href="/asdf-am-signedout" class="govuk-link">Sign in</a> to see your saved pages.
```
What are `/asdf-am-signedin` and `/asdf-am-signedout`?

AlanGabbianelli replied:
@danacotoran can we make it clearer that these are just placeholders using something like
`/placeholder-signedin-endpoint`
and
 `/placeholder-signedout-endpoint`
 ?

AlanGabbianelli commented on app/views/govuk_publishing_components/components/_save_this_page.html.erb:
```diff
@@ -0,0 +1,24 @@
+  signed_in ||= false
+  page_is_saved ||= false
+  page_path ||= nil
+  form_action ||= page_is_saved ? "/my-fancy-delete-action" : "/my-fancy-save-action"
```
Same as the other comment: it would be good to explicitly include the word `placeholder` to avoid confusion.

danacotoran commented:
thank you @AlanGabbianelli I've made some changes:
- renamed the placeholders to contain the word "placeholder" to make it clearer as you suggested
- moved the placeholders and some of the conditional logic to a helper to declutter the partial a bit (not 💯 sure I did it right though)

Will be adding some description text for this PR too, before I un-draft it.

AlanGabbianelli commented on lib/govuk_publishing_components/presenters/save_this_page_helper.rb:
```diff
@@ -0,0 +1,26 @@
+        "sign_in" => "/placeholder-sign-in",
+      }.freeze
+
+      def initialize(options)
+        already_saved = options[:page_is_saved]
+
+        @form_action = already_saved ? FORM_ACTIONS["delete"] : FORM_ACTIONS["save"]
+        @heading_text = already_saved ? I18n.t("components.save_this_page.page_was_saved_heading") : I18n.t("components.save_this_page.page_not_saved_heading")
+        @button_text = already_saved ? I18n.t("components.save_this_page.remove_page_button") : I18n.t("components.save_this_page.add_page_button")
+        @additional_text = options[:signed_in] ? I18n.t("components.save_this_page.see_saved_pages_signed_in", link: LINKS["see_saved_pages"]) : I18n.t("components.save_this_page.see_saved_pages_signed_out", link: LINKS["sign_in"])
+      end
```
I think the initializer can just be
```ruby
attr_reader :page_is_saved, :signed_in
# ...
def initialize(options)
  @page_is_saved = options[:page_is_saved]
  @signed_in = options[:signed_in]
end
```
and all the rest can be extracted into public methods (so you can also unit test them).
